server/container_create.go

```diff
@@ -157,6 +157,10 @@ func buildOCIProcessArgs(containerKubeConfig *pb.ContainerConfig, imageOCIConfig
 func (s *Server) CreateContainer(ctx context.Context, req *pb.CreateContainerRequest) (res *pb.CreateContainerResponse, err error) {
 	logrus.Debugf("CreateContainerRequest %+v", req)
 	s.Update()
+
```

Comment:
Is it a problem that we're acquiring/releasing the same lock along this codepath?

CreateContainer -> Update -> update (acquire, release)
CreateContainer -> acquire, release again

Is this safe if a different thread manages to acquire the lock after update but before the CreateContainer body has run?